tmt/steps/provision/local.py

```diff
@@ -84,7 +84,18 @@ def reboot(self, hard=False):
         return False
 
     def push(self, source=None, destination=None, options=None):
-        """ Nothing to be done to push workdir """
+        """ Push files to the guest """
```

Comment:
I'd like to bring the attention of @psss or @lukaszachy to this particular addition, as I for one am not sure whether replacing no-op implementation with this wouldn't have unexpected side effects. IIUIC, the `push()` can be used to copy various files to guests, like reboot scripts, and the original "do nothing" implementation seems like "let's not change localhost at all" policy - so your change could affect other places that try to upload files but did not succeed with `local` plugin.

Reply:
Hm, we can't push workdir (source and destination will be same).  Personally I'd make 'local' not working (as it is with the reboot) and fix it later - It might need to use sudo.

For testing provision -h container can be used.

Reply:
>  Personally I'd make 'local' not working (as it is with the reboot) and fix it later - It might need to use sudo.

Hi @lukaszachy, if guest is `localhost`, do nothing. But I don't understand `... and fix it later - It might need to use sudo`.  Do you mean we `sudo cp <SRC repo file> /etc/yum.repos.d/` later by adding a method (e.g. `local_execute()`) in `class Guest()`?

BTW, currently [provision](https://tmt.readthedocs.io/en/stable/spec/plans.html#provision) supports four types of guest - `virtual`, `connect`, `container` and `local`. @happz, @lukaszachy and @psss , we should support all types by deploying `.repo` files in guest, right? Please help to confirm, thanks!

Reply:
Maybe a two-level interface could help - according to comment, "pushing workdir" to localhost is undesirable, so we could add `push_workdir()` method which would use `push()` to do the work. `push_workdir()` would remain no-op, leaving `push()` itself working for other stuff plugins might actually like to see materialize even on localhost.